packages/contracts-bedrock/src/cannon/MIPS.sol

```diff
@@ -54,11 +54,15 @@ contract MIPS {
     uint32 constant EINVAL = 0x16;
 
     /// @notice The preimage oracle contract.
-    IPreimageOracle public oracle;
+    IPreimageOracle internal immutable ORACLE;
```

Comment:
This would imply that all chains would use the same mips contract

Reply:
The consequence, IIUC, is mips contracts across all chains have the same codehash? This is fine though.

Reply:
The MIPS contract should be completely stateless, operating as a _pure_ function. It works the same way across all networks. Having an immutable oracle baked into the contract gets us there.